corehq/apps/app_manager/views/modules.py

```diff
@@ -797,29 +807,55 @@ def overwrite_module_case_list(request, domain, app_id, module_unique_id):
                     src_module.case_type)
             )
         else:
-            _update_module_case_list(request, detail_type, src_module, dest_module)
-            dest_modules_msg.append(dest_module.default_name())
-            app.save()
-    _msg = _(f'Case list configuration updated from {src_module.default_name()} menu to {", ".join(map(str, dest_modules_msg))} menu(s).')
+            try:
+                _update_module_case_list(detail_type, src_module, dest_module, attrs_dict)
+                updated_modules.append(dest_module.default_name())
+                app.save()
+            except Exception as e:
+                logger.exception(f'Error in updating module: {dest_module.default_name()}', e)
+                not_updated_modules.append(dest_module.default_name())
+
+    if not_updated_modules:
+        _error_msg = _("Failed to overwrite case lists to menu(s): {}.").format(
+            ", ".join(map(str, not_updated_modules)))
+        messages.error(request, _error_msg)
+
+    _msg = _('Case list configuration updated from {} menu to {} menu(s).').format(
+        src_module.default_name(), ", ".join(map(str, updated_modules)))
     messages.success(request, _msg)
     return back_to_main(request, domain, app_id=app_id, module_unique_id=module_unique_id)
 
-def _update_module_case_list(request, detail_type, src_module, dest_module):
-    display_properties = request.POST.get('display_properties')
-    case_list_filter = request.POST.get('case_list_filter')
-    other_configuration = request.POST.get('other_configuration')
+
+def _validate_overwrite_request(request, detail_type, dest_modules, attrs_dict):
```

Comment:
I'd consider making this return a list of errors, or `None` if the validation is successful, and then updating `messages` in the calling code, because functions are easier to understand when they return data instead of acting via side effects.